### PR TITLE
feat(group): Add GroupShare model

### PR DESCRIPTION
**Pull request by mattrobenolt**
This is the migration split out of #6295

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR includes migrations</td>
    </tr>
  </tbody>
</table>

## Migration Checklist

- [ ] new columns need to be nullable (unless table is new)
- [ ] migration with any new index needs to be done concurrently
- [ ] data migrations should not be done inside a transaction
- [ ] before merging, check to make sure there aren't conflicting migration ids


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/models/groupshare.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+class GroupShare(Model):
+    """
+    A Group that was shared publicly.
+    """
+    __core__ = False
+
+    project = FlexibleForeignKey('sentry.Project')
+    group = FlexibleForeignKey('sentry.Group', unique=True)
+    guid = models.CharField(max_length=32, unique=True, default=lambda: uuid4().hex)
```
Comment (mitsuhiko) [inline src/sentry/models/groupshare.py]: can we call this uuid for consistency with the rest of the codebase that already uses uuid? (eg: dsyms, symcaches, etc.)

guid is a microsoft term that is used for a subset of uuids.

**Review by mitsuhiko — CHANGES_REQUESTED** [review]

File: src/sentry/models/groupshare.py (hunk 1 above)
Comment (mattrobenolt) [reply src/sentry/models/groupshare.py]: We definitely use `guid` in our codebase too inconsistently. `ScheduledDeletion` for example and within `deletions` code we use guid as well as new analytics stuff.

So I agree, but I think we're just inconsistent on both sides and don't think it's worth picking a side here unless we were to change everything.

Comment (mattrobenolt) [reply src/sentry/models/groupshare.py]: So looking, I see exactly 1 model with a `uuid` column, and exactly 1 with `guid`. :)

But I didn't know that GUID was specifically a Microsoft term. So I'll change this to uuid so I feel better about it.

Comment (mattrobenolt) [reply src/sentry/models/groupshare.py]: Done e53ba536ee598da49b13d1812bc6f8c2235d6312

**Review by mitsuhiko — APPROVED** [review]

**Pull request merged**